godel_process_planning/src/blend_process_planning.cpp

```diff
@@ -0,0 +1,157 @@
+#include <godel_process_planning/godel_process_planning.h>
+
+#include <ros/console.h>
+
+// descartes
+#include "descartes_trajectory/axial_symmetric_pt.h"
+#include "descartes_trajectory/joint_trajectory_pt.h"
+#include "descartes_planner/dense_planner.h"
+
+#include "common_utils.h"
+
+
+namespace godel_process_planning
+{
+
+/**
+ * @brief toDescartesPt
+ * @param pose
+ * @return
+ */
+static inline descartes_core::TrajectoryPtPtr toDescartesPt(const Eigen::Affine3d& pose, double dt)
+{
+  using namespace descartes_trajectory;
+  using namespace descartes_core;
+  const descartes_core::TimingConstraint tm (dt);
+  return TrajectoryPtPtr(new AxialSymmetricPt(pose, M_PI/12.0, AxialSymmetricPt::Z_AXIS, tm)); // default timing
+}
+
+static inline double costFunction(const std::vector<double>& source,
+                                  const std::vector<double>& target)
+{
+  double cost = 0.0;
+  for (std::size_t i = 0; i < source.size(); ++i)
+  {
+    double diff = std::abs(source[i] - target[i]);
+    if (diff > M_PI_2) cost += 5.0 * diff;
+    else cost += diff;
+  }
+  return cost;
+}
+
+/**
+ * @brief toDescartesTraj
+ * @param ref
+ * @param points
```

Comment:
Adding comments to these parameter descriptions would be nice, also a field for the _params_ argument is missing.